### PR TITLE
Fix/#1

### LINKED ISSUE #1: setItem, removeItem have undeclared effects
The storage seems to be mutated

### PR DESCRIPTION
**Pull request by philopon**
Hi!

I fixed #1 and its related updates.

Regards.

### CODE REVIEW

**Comment by joneshf** [comment]
Thanks! Just have a couple of questions.

File: src/Browser/WebStorage.purs [hunk 1]
```diff
@@ -12,30 +14,35 @@ module Browser.WebStorage
...
-    key :: s -> Number -> Maybe String
-    length :: s -> Number
-    removeItem :: s -> String -> s
-    setItem :: forall v. s -> String -> v -> s
+    clear :: forall eff. s -> EffWebStorage eff Unit
+    getItem :: forall eff. s -> String -> EffWebStorage eff (Maybe String)
+    key :: forall eff. s -> Number -> EffWebStorage eff (Maybe String)
+    length :: forall eff. s -> EffWebStorage eff Number
+    removeItem :: forall eff. s -> String -> EffWebStorage eff Unit
+    setItem :: forall eff. s -> String -> String -> EffWebStorage eff Unit
```
Comment (joneshf) [inline src/Browser/WebStorage.purs:30]: Why the change to return `Unit`? Is it because the api doesn't return anything to begin with? I'm fine with the change. Just wanted to understand it.

Comment (philopon) [inline src/Browser/WebStorage.purs:30]: Well... I'm not following meaning of its return value, so I deleted it.
actually, Eff can execute side effect without return value. e.g. [`Debug.Trace.trace`](https://github.com/purescript/purescript/blob/master/prelude/prelude.purs#L990-L998)(and I forgot `return {}`. I'll add it soon.).

My rewritten api can use:

```
module Main where

import Control.Monad.Eff
import Browser.WebStorage
import Debug.Trace
import Data.Maybe

main = do
  length localStorage >>= print
  setItem localStorage "foo" "foo"
  setItem localStorage "bar" "bar"
  getItem localStorage "foo" >>= print
  key localStorage 0 >>= print
  length localStorage >>= print
  removeItem localStorage "foo"
  getItem localStorage "foo" >>= print
  clear localStorage
  length localStorage >>= print
```

I think it works well.
But if I don't understand value of return value, please tell me. I'll do rollback it.

**Comment by rudolf-adamkovic** [comment]
Still not merged. :(

**Comment by joneshf** [comment]
Sorry, I don't remember why this didn't get merged...

**Comment by joneshf** [comment]
New release is up. Sorry for the delay.

**Comment by philopon** [comment]
Don't worry. Thanks for merging:smile:
